common/src/test/scala/explore/utils/StreamSpec.scala

```diff
@@ -58,3 +58,23 @@ class StreamSpec extends munit.CatsEffectSuite:
 
     TestControl.executeEmbed(program).map((a, b) => assertNotEquals(a, b))
   }
+
+  test("keyedSwitchEvalMap cancels selectively") {
+    val stream                 = Stream(1, 2, 3).evalMap(IO.sleep(0.25.seconds).as(_))
+    val effect: Int => IO[Int] = i => IO.sleep(1.second) >> IO.pure(i)
+    val pipe                   = keyedSwitchEvalMap[IO, Int, Int, Int](_ % 2, effect)
+
+    val program = stream.through(pipe).compile.toVector
+
+    TestControl.executeEmbed(program).assertEquals(Vector(2, 3))
```

Comment:
I don't really understand what is being asserted here. The effect sleeps for 1 second and emits the same element. So the effect is _always_ cancelled? How is it tested that it is being cancelled? If the effect raises an error or some other noticeable effect, that would be clearer, I think.

I think overall, the tests don't need to do anything with time. I'll admit the `List[Int]` from my tests also aren't super clear. But from a glance I don't really understand what is happening. Maybe they could be clearer if you, for example, have a `Ref[List[String]]` and the effect appends starting, closing and cancelling strings for each element. That way the assertions are clearer on _what_ behaviour is happening and in what order

Reply:
I added some comments to the test. I might still be missing something.